### PR TITLE
add support for managing ssh keys

### LINKED ISSUE #45: Feature request: allow adding ssh keys via the API
The provider currently has no way of adding SSH keys to your Scaleaway account.

Funnily enough, the official Scaleway [docs don't have instructions](https://developer.scaleway.com/#user-data), but there is this community post: 

https://community.online.net/t/api-account-ssh-keys/582/2

```
curl --header 'Content-Type: application/json' \
  --header 'X-Auth-Token: <YOUR TOKEN>' \
  https://account.cloud.online.net/users/<YOUR USER ID> \
  -X PATCH \
  -d '{"ssh_public_keys": [{"key": <YOUR KEY>}, {"key": <YOUR SECOND KEY>}]}'
```

### PR DESCRIPTION
**Pull request by nicolai86**
this PR adds support for managing user ssh keys on scaleway.

the new resource looks like this:

```
resource "scaleway_ssh_key" "awesome_key" {
	key = "ssh-rsa <your key>"
}
```

The implementation relies on `golang.org/x/crypto/ssh` to parse the SSH key to retrieve the correct fingerprint; also included are docs.

SSH keys can be imported like this:

```
terraform import scaleway_ssh_key.id d1:4c:45:59:a8:ee:e6:41:10:fb:3c:3e:54:98:5b:6f
```

closes #45

### CODE REVIEW

File: scaleway/resource_ssh_key.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		})
+	}
+
+	// remote already contains the key, nothing to do
+	if exists {
+		d.SetId(fingerprint)
+		return nil
+	}
+
+	mu.Lock()
```
Comment (tombuildsstuff) [inline scaleway/resource_ssh_key.go]: do you want to move this lock up / re-retrieve the list post-lock, in-case a secondary key gets added in the meantime?

File: scaleway/resource_ssh_key.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+	keys := []api.KeyDefinition{}
+	for _, key := range user.SSHPublicKeys {
+		if !strings.Contains(key.Fingerprint, d.Id()) {
+			keys = append(keys, api.KeyDefinition{
+				Key: key.Key,
+			})
+		}
+	}
+	mu.Lock()
```
Comment (tombuildsstuff) [inline scaleway/resource_ssh_key.go]: again - do we want to move this lock up?

File: scaleway/resource_ssh_key_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		return nil
+	}
+
+	return nil
+}
+
+var testAccCheckScalewaySSHKeyConfig = `
+resource "scaleway_ssh_key" "test" {
+	key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDYpDmIzRs5c+xs0jmljMbNYVcgV8fRruMCRDA4HKjGN2lqLTZhngGDXsdt/2kTNQQPAq2sR4N8mfX5wMRT/+jNb+8esPyY5WlElni0zmD7oLoPW4lYRES6f7EeAv6NttLfkDO42r15OtMnglcgWk1u4o3lOXuLbhzJT1qdicpDja22X3uR/xUy1AYhKBOoiSlQbkb7NhL0lA1xQNwerdaJJS8tFB+wViVDyP0f1HaIRxViFlTGuTbTuIJNR/7VJ9VBBuTnYXaRkPxz64sUXrtdVK8U0+4KsisyXwmgQKnvZBDj91wxz12OOzFSQ52iFprIj1JbkzuBmNWXUGKYzXJZ test"
+}
```
Comment (tombuildsstuff) [inline scaleway/resource_ssh_key_test.go]: given the comments above, it might be worth a test adding multiple keys?

File: vendor/vendor.json [hunk 4]
```diff
@@ -593,6 +593,30 @@
...
+		},
+		{
+			"checksumSHA1": "LXFcVx8I587SnWmKycSDEq9yvK8=",
+			"path": "golang.org/x/crypto/ed25519/internal/edwards25519",
+			"revision": "9de5f2eaf759b4c4550b3db39fed2e9e5f86f45c",
+			"revisionTime": "2018-02-11T11:39:43Z"
+		},
+		{
+			"checksumSHA1": "hfABw6DX9B4Ma+88qDDGz9qY45s=",
+			"path": "golang.org/x/crypto/internal/chacha20",
```
Comment (tombuildsstuff) [inline vendor/vendor.json]: assuming `go build` is happy with the `internal` vendors :)

Comment (nicolai86) [reply vendor/vendor.json]: 👍 it is

File: scaleway/resource_ssh_key.go (hunk 1 above)
Comment (nicolai86) [reply scaleway/resource_ssh_key.go]: done!

File: scaleway/resource_ssh_key.go (hunk 2 above)
Comment (nicolai86) [reply scaleway/resource_ssh_key.go]: done :)

File: scaleway/resource_ssh_key_test.go (hunk 3 above)
Comment (nicolai86) [reply scaleway/resource_ssh_key_test.go]: added testcases for multiple keys including removal.

**Pull request merged**